### ISSUE TITLE
406 when making a request without the json-api media type

### ISSUE THREAD

**Issue by jonnii** [joukevandermaas/saule#100]
Hello! It's really frustrating to test any API using saule in a browser as it will 406 unless the content type matches exactly the json-api media type. I'm unsure as to why you made the decision as my reading of the Content Negotiation section of the spec (http://jsonapi.org/format/#content-negotiation) doesn't seem to indicate that this is required?

It _does_ say that if the content is supplied how it should be handled, but doesn't seem to go further than that.

**Comment by NullVoxPopuli** [comment]
it says this though:

```
Clients MUST send all JSON API data in request documents with the header Content-Type: application/vnd.api+json without any media type parameters.
```

is that what you are not wanting?

**Comment by jonnii** [comment]
The way I read that is is that if the content-type application/vnd.api+json is specified then it must not contain any media type parameters. Perhaps I'm wrong... 

Either way not being able to use a browser to view the API is kind of awkward. The fact that saule also overwrites all the existing formatters and that `PreprocessingDelegatingHandler` is internal means it's not easy to setup the formatting in such a way to fit into an existing application which might also serve json and/or other content types.

**Comment by jonnii** [comment]
And, one more thing. If you try to use it with something like swagger (to allow you to view and test your API) then it does weird things as the other formatters have been removed. It would be great if we could use saule and swagger in the same project.

**Comment by NullVoxPopuli** [comment]
Yeah, there is def room for flexability with this. I'd like something similar to active_model_serializers.

have you used postman? that allows setting the content-type

**Comment by joukevandermaas** [comment]
I definitely agree you should be able to serve other content types from a project that uses Saule. Maybe an overload to `ConfigureJsonApi` that inserts the json api formatter at the start of the collection instead?

**Comment by jonnii** [comment]
swagger lets you set (and even detects) the media-type compatible with your API, it's pretty advanced. i can see the similarity to AMS, it's definitely a good approach! Right now we are returning anonymous types from our controllers, but there's a lot of boilerplate =(

**Comment by joukevandermaas** [comment]
#101 should make configuration more flexible when it's merged. 

I'm not 100% sure on how to read the spec regarding the original issue, but I think we shouldn't change behavior until we get clarification from the authors. Feel free to open an issue over at [the spec's repo](https://github.com/json-api/json-api).

**Comment by jonnii** [comment]
There's already an issue discussing this:

https://github.com/json-api/json-api/issues/968

**Comment by joukevandermaas** [comment]
In the meantime #101 was merged, so half of this issue is resolved. After reading json-api/json-api#968 I agree that the behavior in Saule should be changed. Feel free to open a PR to implement those changes.

**Comment by joukevandermaas** [comment]
Fixed with #151.
